--- ngx_http_sticky_misc.c	2024-08-26 11:36:22.477058230 +0900
+++ ngx_http_sticky_misc.c.new	2024-08-26 13:39:34.643161844 +0900
@@ -17,15 +17,22 @@
 #endif
 
 /* - fix for 1.11.2 removes include <openssl/md5.h> in ngx_md5.h */
-#define MD5_CBLOCK  64
-#define MD5_LBLOCK  (MD5_CBLOCK/4)
-#define MD5_DIGEST_LENGTH 16
-#define SHA_CBLOCK 64
+#ifndef MD5_CBLOCK
+    #define MD5_CBLOCK  64
+    #define MD5_LBLOCK  (MD5_CBLOCK/4)
+#endif
+
+#ifndef MD5_DIGEST_LENGTH
+    #define MD5_DIGEST_LENGTH 16
+#endif
+
+#ifndef SHA_CBLOCK
+    #define SHA_CBLOCK 64
+#endif
 #define SHA_DIGEST_LENGTH 20
 
 #ifndef SHA_DIGEST_LENGTH
-#define SHA_CBLOCK 64
-#define SHA_DIGEST_LENGTH 20
+    #define SHA_DIGEST_LENGTH 20
 #endif
 
 // /* - bugfix for compiling on sles11 - needs gcc4.6 or later*/
