--- ngx_http_sticky_module.c	2024-08-26 11:36:22.477058230 +0900
+++ ngx_http_sticky_module.c.new	2024-08-26 13:39:43.731133872 +0900
@@ -240,7 +240,7 @@
 	ngx_http_set_ctx(r, iphp, ngx_http_sticky_module);
 
 	/* check weather a cookie is present or not and save it */
-	if (ngx_http_parse_multi_header_lines(&r->headers_in.cookies, &iphp->sticky_conf->cookie_name, &route) != NGX_DECLINED) {
+	if (ngx_http_parse_multi_header_lines(r, r->headers_in.cookie, &iphp->sticky_conf->cookie_name, &route) != NULL) {
 		/* a route cookie has been found. Let's give it a try */
 		ngx_log_debug(NGX_LOG_DEBUG_HTTP, r->connection->log, 0, "[sticky/init_sticky_peer] got cookie route=%V, let's try to find a matching peer", &route);
 
@@ -472,7 +472,7 @@
 	}
 
 	if (ctx->sticky_conf->transfer_cookie) {
-		if (ngx_http_parse_set_cookie_lines(&r->upstream->headers_in.cookies, &ctx->sticky_conf->cookie_name, &transfer_cookie) == NGX_DECLINED)
+		if (ngx_http_parse_set_cookie_lines(r, r->headers_in.cookie, &ctx->sticky_conf->cookie_name, &transfer_cookie) == NULL)
 		{
 			ngx_str_null(&transfer_cookie);
 		}
